### PR TITLE
conversion operator is written for table accessor

### PR DESCRIPTION
**Pull request by rezarastak**
Addresses the first part of #8268

### CODE REVIEW

**Comment by masterleinad** [comment]
Would you mind [writing a test](https://www.dealii.org/current/developers/testsuite.html#layoutaddtests) and a changelog entry?

**Comment by drwells** [comment]
This looks good to me. We do need a test though: it can be as simple as filling a table with `std::iota`.

**Comment by rezarastak** [comment]
The error message for the CI test (gcc-serial) seem to come from `grid_in.cc` although there are no `FullMatrix` or `Table` classes in `grid_in.cc`. So I don't know how the changes in this PR are causing these errors.

**Comment by drwells** [comment]
@rezarastak I think the test failures are due to something that was broken and then fixed elsewhere. Would you mind rebasing onto master to update this branch? That should fix everything quickly.

**Comment by drwells** [comment]
The tests still fail due to a lack of disk space.

File: doc/news/changes/minor/20190524RezaRastak [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fix: FullMatrix can now be packed into std::vector and unpacked from it using std::copy and the begin() and end() iterators
```
Comment (masterleinad) [inline doc/news/changes/minor/20190524RezaRastak]:
```suggestion
Fixed: FullMatrix can now be packed into std::vector and unpacked from it using
std::copy and the begin() and end() iterators.
```

File: include/deal.II/base/table.h [hunk 2]
```diff
@@ -1042,6 +1047,11 @@ namespace MatrixTableIterators
      */
     value_type &
     value() const;
+
+    /**
+     * Conversion operator that returns a reference to the element
```
Comment (masterleinad) [inline include/deal.II/base/table.h]:
```suggestion
     * Conversion operator that returns a reference to the element.
```

File: include/deal.II/base/table.h [hunk 3]
```diff
@@ -918,6 +918,11 @@ namespace MatrixTableIterators
     const value_type &
     value() const;
 
+    /**
+     * Conversion operator that returns a constant reference to the element
```
Comment (masterleinad) [inline include/deal.II/base/table.h]:
```suggestion
     * Conversion operator that returns a constant reference to the element.
```

File: tests/lac/matrices_pack_unpack.cc [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2018 by the deal.II authors
```
Comment (masterleinad) [inline tests/lac/matrices_pack_unpack.cc]:
```suggestion
// Copyright (C) 2019 by the deal.II authors
```

**Comment by rezarastak** [comment]
Thank you for the review. I applied the changes you requested.

**Pull request merged**